runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/util/CloudObjectTranslator.java

```diff
@@ -34,4 +34,10 @@
    * Converts back into the original object from a provided {@link CloudObject}.
    */
   T fromCloudObject(CloudObject cloudObject);
+
+  /**
+   * Gets the class name that will represent any {@link CloudObject} created by this {@link
+   * CloudObjectTranslator}.
+   */
+  String cloudObjectClassName();
```

Comment:
Typically you would have the registrar provide the mapping of string -> translator and class -> translator and not have the translator self report half of the mapping.